lib/spack/spack/util/windows_registry.py

```diff
@@ -59,18 +59,88 @@ def subkeys(self):
     def hkey(self):
         return self._handle
 
+    def OpenKeyEx(self, subname, **kwargs):
+        """Convenience wrapper around winreg.OpenKeyEx"""
+        tty.debug(
+            f"[WINREG ACCESS] Accessing Reg Key {self.path}/{subname} with {kwargs.get('access', 'default')} access"
+        )
+        try:
+            return winreg.OpenKeyEx(self.hkey, subname, **kwargs)
+        except OSError as e:
+            # Expected errors that occur on occasion, these are easily
+            # debug-able and have sufficiently verbose reporting and obvious cause
+            if hasattr(e, "winerror") and e.winerror in (5, 2):
+                raise e
+            # Other OS errors are more difficult to diagnose, so we wrap them in some extra
+            # reporting
+            raise InvalidRegistryOperation("OpenKeyEx", e, self.path, subname, **kwargs) from e
```

Comment:
I see some repetition between this and `EnumKey`/`EnumValue`: would it be worth collecting this logic into a common object with a `__call__` method? This might be a naive observation, and e.g. if the reused error codes are circumstantial then I could see the value in keeping them separate (possibly with an extra comment though).

Reply:
It's generalizable, I think another method in the class implementing this logic is probably sufficient.